### DISCUSSION TITLE
oidc without keycloak

### DISCUSSION THREAD

**Discussion by maxkomarychev** [wundergraph/cosmo#681] [Q&A]
Is there a documented way to setup studio with OIDC provider without using keycloak?

Is this an expected experience in on-prem deployment?

<img width="1466" alt="image" src="https://github.com/wundergraph/cosmo/assets/7002833/e3c05f9f-da8b-49ae-be5c-12cfe89f9a3a">

**Comment by StarpTech** [comment]
Hi @maxkomarychev, keycloak is required and responsible for all authentication related features.
